test/utils_test.js

```diff
@@ -306,6 +307,210 @@ describe("utils", () => {
+          level1: { level2: ["child-default-1", "root-default-2"] },
+        });
+      });
+
+      it("should overwrite schema array item defaults from parent for nested default definitions", () => {
+        const schema = {
+          type: "object",
+          default: {
+            level1: {
+              level2: [{ item: "root-default-1" }, { item: "root-default-2" }],
+            },
+          },
+          properties: {
+            level1: {
+              type: "object",
+              default: { level2: [{ item: "parent-default-1" }, {}] },
+              properties: {
+                level2: {
+                  type: "array",
+                  items: {
+                    type: "object",
+                    properties: {
+                      item: {
+                        type: "string",
+                      },
+                    },
+                  },
+                },
+              },
+            },
+          },
+        };
+
+        expect(getDefaultFormState(schema, {})).eql({
+          level1: { level2: [{ item: "parent-default-1" }, {}] },
+        });
+      });
+
+      it("should merge schema array item defaults from sam item for overlapping default definitions", () => {
+        const schema = {
+          type: "object",
+          properties: {
+            level1: {
+              type: "array",
+              default: ["property-default-1", "property-default-2"],
+              items: [
+                {
+                  type: "string",
+                  default: "child-default-1",
+                },
+                {
+                  type: "string",
+                },
+              ],
+            },
+          },
+        };
+
+        expect(getDefaultFormState(schema, {})).eql({
+          level1: ["child-default-1", "property-default-2"],
+        });
+      });
+
+      it("should merge schema from additionalItems defaults into property default", () => {
+        const schema = {
+          type: "object",
+          properties: {
+            level1: {
+              type: "array",
+              default: [
+                {
+                  item: "property-default-1",
+                },
+                {},
+              ],
+              additionalItems: {
+                type: "object",
+                properties: {
+                  item: {
+                    type: "string",
+                    default: "additional-default",
+                  },
+                },
+              },
+              items: [
+                {
+                  type: "object",
+                  properties: {
+                    item: {
+                      type: "string",
+                    },
+                  },
+                },
+              ],
+            },
+          },
+        };
+
+        expect(getDefaultFormState(schema, {})).eql({
+          level1: [
+            { item: "property-default-1" },
+            { item: "additional-default" },
```

Comment:
Is this an intuitive / logical result? (If a schema author purposefully makes the second item in the array default `{}`, then the second item is injected by the default of `additionalItems`)

Reply:
It's in line with merging of objects in defaults and merging of array with the form data.
It's logical from the point that we try to merge as much deeply as we possible can. I tried to keep the special rules about merging as small as possible here.

Reply:
Oh, makes sense -- I missed that `items` has a length of one here, so it makes sense for the default from `additionalItems` to apply to any items after the first one.

Can you add another test with the same schema, except for with `items` of length two, and thus, it should _not_ overwrite the default value of item 2?